### ISSUE TITLE
VLAN Assignment with Nested Locations

### ISSUE THREAD

**Issue by zsblevins** [nautobot/nautobot#6417]
### Environment
* Nautobot version (Docker tag too if applicable): 2.3.4
* Python version: 3.10
* Database platform, version: Postgres
* Middleware(s):

### Steps to Reproduce
1. Create a new Location Type (e.g. Module) with parent location type Site
2. Create a Site location (e.g. SiteA)
3. Create a Module location under SiteA (e.g. DataHall 1)
4. Create a Device and assign location to DataHall 1
5. Create a VLAN and associate it to SiteA
6. Create an interface on the Device from step 4 and assign untagged VLAN from step 5

### Expected Behavior
I would expect the allocation to succeed as the device is assigned to a location nested within the Site associated to the VLAN

### Observed Behavior
```
swp18: The request failed with code 400 Bad Request: {'untagged_vlan': ["The untagged VLAN (Network Management (30)) must have a common location as the interface's parent device, or it must be global."]}
```

Note that this is painful for us as we track rooms/data halls/etc as another layer below the site and allocate our racks/devices based on which room they're installed in. As a result, when we add a VLAN definition for a site, we also have to add the association for all of the rooms within that site, which can be tedious.

**Comment by glennmatthews** [comment]
This was enhanced recently by #6379, which is in v2.3.8. Can you upgrade to 2.3.8 and confirm whether the new behavior is sufficient to close this issue?

**Comment by zsblevins** [comment]
Confirmed it appears fixed in 2.3.8, apologies for the duplicate issue, thanks!
